beacon-chain/blockchain/service.go

```diff
@@ -226,6 +196,22 @@ func (c *ChainService) updateHead(slot uint64) {
 	c.candidateCrystallizedState = nilCrystallizedState
 }
 
+// doesPoWBlockExist checks if the referenced PoW block exists.
+// If the node is not connected to a PoW chain, defaults to true.
+func (c *ChainService) doesPoWBlockExist(block *types.Block) bool {
+	if c.web3Service == nil {
+		return true
+	}
+
+	fetcher := c.web3Service.Client()
+	if _, err := fetcher.BlockByHash(context.Background(), block.PowChainRef()); err != nil {
```

Comment:
Shouldn't we check the response? I'm wondering if some web3 API implementations return nil rather than returning an error if a block is not present.

Reply:
[here's the original line](https://github.com/prysmaticlabs/prysm/blob/db1138ae1f14df62b512ed17d5f97308db0a685c/beacon-chain/blockchain/core.go#L189). The method returns an error if the block doesn't exist. I guess `fetcher.blockExists()` would be a friendly interface for this line.

Reply:
Actually, you're right. [here's the web3 api](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbyhash). It returns nil if the block doesn't exist, I'll fix this.